File: Arrays/Algorithms/Miscellaneous/majority_element.py
def majority_element(nums):
	count = 0
	candidate = None
	for num in nums:
		if count == 0:
			candidate = num
		count += (1 if num == candidate else -1)
	if nums.count(candidate) > len(nums) // 2:
		return candidate
	return -1

if __name__ == "__main__":
	n = int(input())
	nums = list(map(int, input().split()))
	print(majority_element(nums))

File: Arrays/Algorithms/Miscellaneous/majority_element.cpp
#include <bits/stdc++.h>
using namespace std;

int majorityElement(vector<int>& nums) {
	int count = 0, candidate = 0;
	for (int num : nums) {
		if (count == 0) candidate = num;
		count += (num == candidate) ? 1 : -1;
	}
	count = 0;
	for (int num : nums) if (num == candidate) count++;
	return (count > nums.size() / 2) ? candidate : -1;
}

int main() {
	int n;
	cin >> n;
	vector<int> nums(n);
	for (int i = 0; i < n; i++) cin >> nums[i];
	cout << majorityElement(nums) << endl;
	return 0;
}
